driver/mounter.go

```diff
@@ -214,47 +230,41 @@ func (m *mounter) Unmount(target string) error {
 	return mount.CleanupMountPoint(target, m.kMounter, true)
 }
 
-func (m *mounter) IsRunning(source string, cmdContext execContext) bool {
-	cmd := fmt.Sprintf("ls -l %s | awk '{print $NF}'", source)
-	out, err := cmdContext("/bin/sh", "-c", cmd, source).CombinedOutput()
-
+func (m *mounter) IsRunning(av iAttachmentValidator, source string) bool {
+	out, err := av.evalSymlinks(source)
 	if err != nil {
 		m.log.WithFields(logrus.Fields{
-			"cmd": cmd,
-		}).Errorf("error retrieving the mount name, err: %v", err)
+			"source": source,
+		}).Errorf("error evaluating the sym link, err: %v", err)
```

Comment:
The pattern of logging errors and returning `false` only to turn that outcome into another error at the call site seems cumbersome to me. I'd prefer if we just always returned an error with all relevant info contained. For instance, in this case we could return

```go
return fmt.Errorf("failed evaluating symlink %q: %s", source, err)
```

(Side-note: I used `%s` here because we want the string representation of an error with certainty.)

That way, the call site can do the logging centrally.

If we want to follow this path, then I think it'd also make sense to rename the method to something like `EnsureAttached` and let it return an `error` only.

Reply:
Nice ! I've refactored all the errors to use the new proposed format.